### ISSUE TITLE
Markdown syntax highlighter doesn't understand asterisks within backticks

### ISSUE THREAD

**Issue by whereswaldon** [mawww/kakoune#1769]
Perhaps it isn't a widely-enough standardized feature of the 87-billion flavors of markdown, but if you write the following:

```markdown
This is a demo of how referencing a Go type like `*big.Int` causes syntax highlighting problems.
That pointer type causes all of the problems. Curse you `*big.Int`!
```

You get the same highlighting as an *italicized* word between the two asterisks. Below is a screenshot of the text that led me to notice the problem.

Screenshot:
![image](https://user-images.githubusercontent.com/3922125/33915368-f328b428-df70-11e7-886e-1dc8c9b3c983.png)

However, `*big.Int` should (according to GitHub markdown at least) be interpreted as a literal asterisk because it is within backticks.

1. Does the flavor of markdown that Kakoune highlights for support literal asterisks within backticks?
2. Can this be fixed easily?

Thanks for all of the great work on Kakoune!

**Comment by mawww** [comment]
Hello,

This might be fixable by using regions instead of raw regex for those blocks, at least for backticks which introduce a new "context" where no other markdown formatting should take place.

**Comment by danr** [comment]
This is a regression introduced by my commit some days ago, but it is solved with my new PR #1765

**Comment by whereswaldon** [comment]
I can confirm that the issue is solved now. Thanks @danr !
